rust/parser/syntax_error.rs

```diff
@@ -20,7 +20,7 @@
  *
  */
 
-use std::fmt::{Display, Formatter};
+use std::fmt;
```

Comment:
The usage of `std::fmt::Display` is unified across the repo in line with major crates (tokio, chrono).

Reply:
I think that's just to reduce the number of imports.

Reply:
Let's go back to importing `Display` and `Formatter` directly. At the cost of one extra import, we get terser code that we use _very_ regularly.

We frequently take inspiration from other open-source code, review their standards, and then see if these standards are the best possible according to our values. In this case, I believe `Display` and `Formatter` are minimal and non-ambiguous, which is our benchmark for naming.

Reply:
I don't think getting rid of five characters (`fmt::`) is a good enough reason to go against a convention that every other Rust project seems to follow.

Reply:
I had a further look into this. It looks like it is always referred to as `fmt::Display` in the Rust docs. That's probably why other Rust projects refer to it in the same way.

From a good number of crate dependencies I searched, all refer to it as `fmt::Display`. Clearly it is an accepted convention, but we do not blindly follow conventions that are not backed up by good reason.

In this case, it looks like all the methods in the `fmt` module are designed to be namespaced with `fmt`. For example, `fmt::print`, `fmt::LowerHex`. If we pick these apart (the latter in particular) the need for `fmt` prefix becomes clear: we're implementing a lower hex _formatter_, not a "lower hex".

Likewise, when we implement `Display`, we're implementing a Display formatter (ie pretty-print), and when we implement `Debug`, we're implementing a debug formatter. So in conclusion, the convention is a sensible one and we should follow it.

As for why we put in the effort to decide at all - it's weird at a glance because we constantly `#[derive(Debug)]` without a thought despite it actually being `fmt::Debug`, so `Display` should intuitively be in the same boat. But it isn't, because of the context in which `Display` is defined; a formatting library that defines many format helpers, of which `Display` is one. In terms of readability, whether it was 5 or 30 characters barely matters, it's adding a new logical block to the definition of something that we implement for _almost every_ API object we define. The conventions we decide in `typeql` will go on to decide the conventions in our other Rust projects including `typedb`.